Format Java code and enforce formatting in CI

gj opened the pull request:
PR checklist:
- [ ] Update docs example line numbers now that they've shifted in the formatting.

samscott89 commented:
Is it possible to (a) add the check to `make lint` and (b) add the formatting pass to `make fmt` (in the root dir Makefile)?

gj commented:
> Is it possible to add the formatting pass to `make fmt` (in the root dir Makefile)?

Added `make java-fmt` and `make -C languages/java fmt`, and added the former to the `make fmt` task.
